proposals/POEM-2-function-pulling-container-scheduler.md

```diff
@@ -0,0 +1,132 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Title
+Function Pulling Container Scheduler(FPCScheduler)
+
+## Status
+* Current state: Draft
+* Author(s): @style95, @ningyougang, @keonhee, @jiangpengcheng, @upgle
+
+## Summary and Motivation
+
+This POEM proposes a new scheduler for OpenWhisk, FPCScheduler.
+Previously we revealed [performance issues](https://cwiki.apache.org/confluence/display/OPENWHISK/Autonomous+Container+Scheduling+v1) in OpenWhisk.
+There are many reasons for them, but we can summarize them into the following problems.
+
+1. Multiple actions share the same `homeInvoker`.
+2. Current scheduler does not consider time taking for container operations(create/delete, pause/unpause).
+3. Resources are evenly divided by the number of controllers.
+
+First, multiple actions share the same `homeInvoker`. This is because the `homeInvoker` is statically decided by a hash function.
+While scheduling, resource status in the invoker side is not considered and this introduces busy hotspot invokers while the others are idle.
+This is generally a good idea to increase the probability of container reuse, but it becomes bad in the real scene because of slow container operation.
+When all resources in an invoker are taken by one action(busy/warm), an activation for another action will remove one of the existing containers and create a new one for it.
+If actions run for a long time this would be a good approach in terms of bin-packing. But when actions run for short time(most of the serverless cases),
+it severely increases the response time because container operations take one to two orders of magnitude longer time than invocation and it results in bad performance.
+
+Second, since controllers schedule activations to invokers in a distributed environment, controllers are assigned resource fraction of invokers.
```

Comment:
```suggestion
Second, since controllers schedule activations to invokers in a distributed environment, the invokers in the system are partitioned and assigned to specific controllers. This means that any one controller is assigned a fraction of the available resources in the system.
```